amlip_py/amlip_py/node/FiwareNode.py

```diff
@@ -0,0 +1,558 @@
+        else:
+            super().__init__(name, domain)
+
+        self.server_ip = server_ip
+        self.server_port = server_port
+        self.context_broker_ip = context_broker_ip
+        self.context_broker_port = context_broker_port
+        self.id = entity_id
+        self.entity_data = entity_data
+        self.entity_solution = entity_solution
+        self.logger = logger
+        self.logger.info(f'Creating FiwareNode with name: {name}')
+
+        self.name = name
+
+        self.edge = AsyncEdgeNode('edge_node'+name,
+                                  listener=InferenceListenerLambda(self.inference_received))
+
+        self.app = Flask(__name__)
+
+        self.setup_routes()
+        self.init_subscriptions()
+
+    def setup_routes(self):
+        """
+        Handle requests to the /accumulate route.
+
+        This function will be called for all HTTP methods (GET, POST, PUT, DELETE, PATCH).
+
+        """
+
+        self.app.add_url_rule('/accumulate',
+                              view_func=self.accumulate,
+                              methods=['GET', 'POST', 'PUT', 'DELETE', 'PATCH'])
+
+    def init_subscriptions(self):
+        """
+        Initialize subscriptions to notify on all changes to the entity with self.id.
+
+        """
+
+        subscription_data = {
+            'description': 'Notify me of all changes',
+            'subject': {
+                'entities': [
+                    {
+                        'id': self.id,
+                        'type': 'string'
+                    }
+                ],
+                'condition': {
+                    'attrs': [
+                        # make a notification trigger on changes to the self.entity_data attribute
+                        self.entity_data
+                    ],
+                    'notifyOnMetadataChange': False
+                }
+            },
+            'notification': {
+                'http': {
+                    'url': f'http://{self.server_ip}:{self.server_port}/accumulate'
+                },
+                'attrs': []  # all the attributes in the entity
+            }
+        }
+
+        response = requests.post(
+            f'http://{self.context_broker_ip}:{self.context_broker_port}/v2/subscriptions',
+            headers=headers_POST,
+            data=json.dumps(subscription_data))
+
+        response.raise_for_status()
+
+    def run(self):
+        """
+        Start the Flask server.
+
+        """
+
+        self.app.run(host=self.server_ip, port=self.server_port)
+
+    def post_data(
+            self,
+            data):
+        """
+        Post data to the Fiware context broker.
+
+        Parameters
+        ----------
+        data : dict
+            Data to be posted.
+
+        """
+        global waiter_inference
+
+        # __FLAG__
+        # We want to close the waiter when we post data
+        waiter_inference = BooleanWaitHandler(True, False)
+        # __FLAG__
```

Comment:
Instead of initializing the `waiter_inference` in the `post_data` method, it should be initialized in the constructor and then closed each time `post_data` is called. Otherwise, if we call `get_inference()` without having previously called `post_data()`, the waiter wouldn’t exist.

Reply:
Done